DonoGear/app/src/main/java/com/example/donogear/actionpages/MainActivity.java

```diff
@@ -171,6 +293,8 @@ private void initializeLayout() {
         context = getBaseContext();
         listOfItems = new ArrayList<>();
         copyList = new ArrayList<>();
+        donorDetailsList = new ArrayList<>();
+        causesDetailsList = new ArrayList<>();
         superCopyList = new ArrayList<>();
         tags = new ArrayList[2];
```

Comment:
Should add a comment about why the arrayList value of 2 is hardcoded or create a constant for it